### ISSUE TITLE
Count user queries and then prompt big users for contributions

### ISSUE THREAD

**Issue by mlissner** [freelawproject/courtlistener#914]
When people search the RECAP Archive and don't have the extension installed, we bug them to install it with the idea being that if they're querying the DB, they should contribute to it. 

A suggestion today was that we should also prompt people for contributions if they do more than a certain number of queries. It's about awareness that FLP is a non-profit and about getting people to think about whether they can/should contribute. 

I'm tempted to build this out as a search history feature. For some time, I've wanted to keep track of people's queries so that they can find their old work again (and so, perhaps we can pop it into the query box for folks). For now, that's complicated and has privacy issues, so I think we can just keep track of their query *count*, and then prompt using that.

I'd say the best way to do this is to start collecting the per-user counts first, then when we know that, we can think about when to put the prompt. 

It'd also be nice to do this for unauthenticated users, though I don't know if that's easy or worth it.

**Comment by mlissner** [comment]
This was a good idea and we have search history now. Putting this on the backlog.

**Comment by tclancy** [comment]
Would this just be a matter of grouping the `SearchQuery` by user and making it available for display? Would you want to filter by date range at all? Feels like this should be stuck in the cache and invalidated/ incremented when a new `SearchQuery` is made by the user.

**Comment by mlissner** [comment]
Yeah, the search query table is very large, so trying to count the user's number of requests won't perform, but it's not crazy to put it in redis. I think if we did that, we'd want: 

 - A django command or script to backfill redis based on the current searchquery table. 
 - A bit of code to increment the counts as needed (excluding API queries). This might be able to follow the pattern that's used by the API logger, which keeps this for API queries (maybe it'd even integrate to prevent code duplication — that might be a bad idea, but it's worth checking?) 
 - A little alert thing in the search results that pops up according to the milestones we have in the stats module starting at maybe 50 queries? The messages we send could be different based on the number of queries they've made, but with a light touch? And the message shouldn't show up for donors or members.

Then we just check the count when the user does a query, increment it, and if it hits a milestone, we show the appropriate thing to please donate or become a member.

All in all, pretty simple, I think?
